src/Integrations/ActiveCampaign/ActiveCampaignClient.php

```diff
@@ -0,0 +1,675 @@
+	 *
+	 * @return array<string, mixed>
+	 */
+	public function postApplication(string $itemId, array $params, array $files, string $formId): array
+	{
+		// Map body.
+		$requestBody = [
+			'contact' => $this->prepareParams($params),
+		];
+
+		// Make an API request.
+		$response = \wp_remote_request(
+			"{$this->getBaseUrl()}contacts",
+			[
+				'headers' => $this->getHeaders(),
+				'method' => 'POST',
+				'body' => \wp_json_encode($requestBody),
+			]
+		);
+
+		// Output response details.
+		$details = $this->getApiReponseDetails($response);
+		$code = $details['code'];
+		$body = $details['body'];
+
+		// Bailout if wp error.
+		if (\is_wp_error($response)) {
+			return $this->getApiWpErrorOutput(
+				SettingsActiveCampaign::SETTINGS_TYPE_KEY,
+				$details,
+				$requestBody,
+				$response->get_error_message()
+			);
+		}
+
+		switch ($code) {
+			case 200:
+			case 201:
+				return $this->getApiSuccessOutput(
+					SettingsActiveCampaign::SETTINGS_TYPE_KEY,
+					[
+						'contactId' => $body['fieldValues'][0]['contact'],
+					]
+				);
+			case 403:
+				return $this->getApiErrorOutput(
+					SettingsActiveCampaign::SETTINGS_TYPE_KEY,
+					$details,
+					$requestBody,
+					$this->getErrorMsg([
+						[
+							'code' => 'activeCampaignForbiden',
+						]
+					]),
+				);
+			case 500:
+				return $this->getApiErrorOutput(
+					SettingsActiveCampaign::SETTINGS_TYPE_KEY,
+					$details,
+					$requestBody,
+					$this->getErrorMsg([
+						[
+							'code' => 'activeCampaign500',
+						]
+					]),
+				);
+			default:
+				return $this->getApiErrorOutput(
+					SettingsActiveCampaign::SETTINGS_TYPE_KEY,
+					$details,
+					$requestBody,
+					$this->getErrorMsg($body['errors'] ?? []),
+				);
+		}
+	}
+
+	/**
+	 * API request to post tags.
+	 *
+	 * @param string $tag Tag to store.
+	 * @param string $contactId Contact ID to store.
+	 *
+	 * @return array<string, mixed>
+	 */
+	public function postTag(string $tag, string $contactId): array
+	{
+		// Check if tag exist using api.
+		$tagId = $this->getExistingTagId($tag);
+
+		// If tag is missing create new using api.
+		if (!$tagId) {
+			$tagId = $this->createNewTag($tag);
+		}
+
+		// Prepare body.
+		$requestBody = [
+			'contactTag' => [
+				'contact' => $contactId,
+				'tag' => $tagId,
+			],
+		];
+
+		// Make request to map contact with tags.
+		$response = \wp_remote_request(
+			"{$this->getBaseUrl()}contactTags",
+			[
+				'headers' => $this->getHeaders(),
+				'method' => 'POST',
+				'body' => \wp_json_encode($requestBody),
+			]
+		);
+
+		// Output response details.
+		$details = $this->getApiReponseDetails($response);
+		$code = $details['code'];
+		$body = $details['body'];
+
+		// Bailout on wp error.
+		if (\is_wp_error($response)) {
+			return $this->getApiWpErrorOutput(
+				SettingsActiveCampaign::SETTINGS_TYPE_KEY,
+				$details,
+				$requestBody,
+				$response->get_error_message()
+			);
+		}
+
+		switch ($code) {
```

Comment:
This `switch` seems like it's repeating in the previous method, maybe you can extract this part out to a separate method for handling error codes?

Reply:
There are a slight difference between method so I would like to leave it like it is now. I will do some refactoring later on on this